test/kt/single_pass_scan.cpp

```diff
@@ -0,0 +1,236 @@
+                                                           TestUtils::get_new_kernel_params<1>(param))
+            .wait();
+    }
+
+    auto acc = buf_out.get_host_access();
+
+    std::string msg = "wrong results with buffer, n: " + std::to_string(size);
+    EXPECT_EQ_RANGES(ref, acc, msg.c_str());
+}
+#endif
+
+template <typename T, sycl::usm::alloc _alloc_type, typename BinOp, typename KernelParam>
+void
+test_usm(sycl::queue q, std::size_t size, BinOp bin_op, KernelParam param)
+{
+#if LOG_TEST_INFO
+    std::cout << "\t\ttest_usm<" << TypeInfo().name<T>() << ", " << USMAllocPresentation().name<_alloc_type>() << ">("
+              << size << ");" << std::endl;
+#endif
+    std::vector<T> expected(size);
+    generate_scan_data<BinOp>(expected.data(), size, 42);
+
+    TestUtils::usm_data_transfer<_alloc_type, T> dt_input(q, expected.begin(), expected.end());
+    TestUtils::usm_data_transfer<_alloc_type, T> dt_output(q, size);
+
+    std::inclusive_scan(expected.begin(), expected.end(), expected.begin(), bin_op);
+
+    oneapi::dpl::experimental::kt::gpu::inclusive_scan(q, dt_input.get_data(), dt_input.get_data() + size,
+                                                       dt_output.get_data(), bin_op,
+                                                       TestUtils::get_new_kernel_params<2>(param))
+        .wait();
+
+    std::vector<T> actual(size);
+    dt_output.retrieve_data(actual.begin());
+
+    std::string msg = "wrong results with USM, n: " + std::to_string(size);
+    EXPECT_EQ_N(expected.begin(), actual.begin(), size, msg.c_str());
+}
+
+template <typename T, typename BinOp, typename KernelParam>
+void
+test_sycl_iterators(sycl::queue q, std::size_t size, BinOp bin_op, KernelParam param)
+{
+#if LOG_TEST_INFO
+    std::cout << "\t\ttest_sycl_iterators<" << TypeInfo().name<T>() << ">(" << size << ");" << std::endl;
+#endif
+    std::vector<T> input(size);
+    std::vector<T> output(size);
+    generate_scan_data<BinOp>(input.data(), size, 42);
+    std::vector<T> ref(input);
+    std::inclusive_scan(std::begin(ref), std::end(ref), std::begin(ref), bin_op);
+    {
+        sycl::buffer<T> buf(input.data(), input.size());
+        sycl::buffer<T> buf_out(output.data(), output.size());
+        oneapi::dpl::experimental::kt::gpu::inclusive_scan(q, oneapi::dpl::begin(buf), oneapi::dpl::end(buf),
+                                                           oneapi::dpl::begin(buf_out), bin_op,
+                                                           TestUtils::get_new_kernel_params<3>(param))
+            .wait();
+    }
+
+    std::string msg = "wrong results with oneapi::dpl::begin/end, n: " + std::to_string(size);
+    EXPECT_EQ_RANGES(ref, output, msg.c_str());
+}
+
+template <typename T, typename BinOp, typename KernelParam>
+void
+test_general_cases(sycl::queue q, std::size_t size, BinOp bin_op, KernelParam param)
+{
+    test_usm<T, sycl::usm::alloc::shared>(q, size, bin_op, param);
```

Comment:
It's a matter of taste, but using `TestUtils::get_new_kernel_params` here instead of the functions below may be better if we want to add the same at `test_all_cases` level.